literal/utils.go

```diff
@@ -14,8 +17,8 @@ func NewBool(value bool) (expr.Literal, error) {
 	return expr.NewPrimitiveLiteral[bool](value, false), nil
 }
 
-func NewInt8(value int8) (expr.Literal, error) {
-	return expr.NewPrimitiveLiteral[int8](value, false), nil
+func NewInt8(value int8) expr.Literal {
```

Comment:
Nit:  It's way too late but in the future consider using FromInt8 instead as this is a conversion and not the construction of a pristine value.

Reply:
ok, noted!